src/odemis/driver/semcomedi.py

```diff
@@ -928,23 +931,21 @@ def setup_timed_command(self, subdevice, channels, ranges, period_ns,
         cmd.start_arg = start_arg
         cmd.stop_src = stop_src
         cmd.stop_arg = stop_arg
-        # The get_cmd_generic_timed likes to spread the conversion, if the
-        # hardware allows it, but we don't.
-        if (cmd.convert_src == comedi.TRIG_TIMER and
-            cmd.scan_begin_src == comedi.TRIG_FOLLOW):
-            logging.debug("Converting 'follow' mode to 'timer' mode")
-            cmd.scan_begin_src = comedi.TRIG_TIMER
-            cmd.scan_begin_arg = period_ns
-            cmd.convert_src = comedi.TRIG_NOW
-            cmd.convert_arg = 0
 
         self._prepare_command(cmd)
 
-        if (cmd.scan_begin_arg != period_ns or cmd.start_arg != start_arg or
+        if (cmd.convert_src == comedi.TRIG_TIMER and
+            cmd.scan_begin_src == comedi.TRIG_FOLLOW):
+            logging.debug("Using 'follow' mode for conversion")
+            actual_period = cmd.convert_arg * nchans
+        else:
+            actual_period = cmd.scan_begin_arg
+
+        if (actual_period != period_ns or cmd.start_arg != start_arg or
             cmd.stop_arg != stop_arg):
```

Comment:
No. That's actually why this test is there: _prepare_command() will modify cmd to fit it to the actual hardware limitations.